dictionaries/src/main/java/com/anysoftkeyboard/dictionaries/WordComposer.java

```diff
@@ -63,12 +63,21 @@ public void reset() {
     }
 
     /**
-     * Number of keystrokes in the composing word.
+     * Number of keystrokes (codepoints, not chars) in the composing word.
      *
      * @return the number of keystrokes
      */
     @Override
     public int length() {
```

Comment:
I guess a better name for this is `codesCount`?

Reply:
_Thank you._ That's a great idea, having `length()` meaning different things depending on the type of variable is very confusing.

I couldn't find a way remove completely the method `length()` or even print deprecation notices to the logcat, because it's also called on `KeyCodesProvider`-typed variables and... I'm not knowledgeable with Java enough to affect `WordComposer` without affecting `KeyCodesProvider`.

So I chose `codePointCount()` and `charCount()` (named after related functions at [String](https://developer.android.com/reference/java/lang/String.html#codePointCount(int,%20int)) and [Character](https://developer.android.com/reference/java/lang/Character.html#charCount(int))). `length()` is a deprecated synonym for `codePointCount()`, for the moment.

Commit 77f28dc9cbf2c6b62f1d685e656981549a93a389. I forgot to add a comment for future developers, I'll add it now.